src/session.rs

```diff
@@ -965,7 +965,11 @@ pub struct PacketReceipt {
 }
 
 fn get_params(c: &CodecConfig, pt: Pt) -> Option<&PayloadParams> {
-    c.iter().find(|p| p.pt() == pt)
+    if let p @ Some(_) = c.iter().find(|p| p.pt() == pt) {
+        return p;
+    }
+
+    main_payload_type_for(c, pt).and_then(|pt| c.iter().find(|p| p.pt == pt))
```

Comment:
I think you should remove the `get_params` instead and call `main_payload_type_for` directly from the (only) callsite.

Reply:
See c7506b6, I changed it to both remap and find the PayloadParams in one go. No more double traversals over the codecs